spock-specs/src/test/groovy/org/spockframework/smoke/parameterization/MethodParameters.groovy

```diff
@@ -256,6 +256,28 @@ def foo() {
     ]
   }
 
+  @Foo
+  def "method arguments in data driven feature methods can be provided by extensions"(a) {
+    expect:
+    x == 1
+    y == 2
+    a == 'foo'
+
+    where:
+    x = 1
+    y = 2
+  }
+
+  @Foo
+  def "additional method arguments do not influence data values in iteration info"(a) {
```

Comment:
What about interspersed usage, e.g., something like this `(a,b,c)` with `a,c,x,y` being data variable and `b` being provided by an extension.

Reply:
The where block rewriter on compilation adds all missing data variables that are not explicitly declared as parameters and sorts the parameters, so that first all data variable are parameters and in the order of the where block, then all auxiliary parameters in declaration order: `org.spockframework.compiler.WhereBlockRewriter#handleFeatureParameters`.
So even if you declare `(a, b, c)`, after compilation it will be `(a, c, b)`.